app/serialization.py

```diff
@@ -89,7 +91,50 @@ def deserialize_host_xjoin(data):
     return host
 
 
-def serialize_host(host, staleness_timestamps, fields=DEFAULT_FIELDS):
+def serialize_host(host, staleness_timestamps, fields=DEFAULT_FIELDS, sparse_fieldset=None):
+    if sparse_fieldset:
+        serialized_host = _sparse_fieldset_serialization(host, staleness_timestamps, sparse_fieldset)
+    else:
+        serialized_host = {**serialize_canonical_facts(host.canonical_facts)}
+        for field in fields:
+            _serialize_host_field(host, field, staleness_timestamps, serialized_host)
+
+    return serialized_host
+
+
+def _sparse_fieldset_serialization(host, staleness_timestamps, sparse_fieldset):
+    serialized_host = {}
+
+    if "host" in sparse_fieldset:
+        host_attributes = sparse_fieldset["host"].replace(" ", "").split(",")
+        canonical_facts_attributes = list(set(host_attributes) & set(_CANONICAL_FACTS_FIELDS))
+        host_attributes = list(set(host_attributes) - set(canonical_facts_attributes))
+        serialized_host = {
+            **serialize_canonical_facts(host.canonical_facts, canonical_fields=canonical_facts_attributes)
+        }
+        for host_attribute in host_attributes:
+            _serialize_host_field(host, host_attribute, staleness_timestamps, serialized_host)
```

Comment:
Something like this should work:

```suggestion
        serialized_host = {
            **serialize_canonical_facts(host.canonical_facts, canonical_facts_attributes),
            **_serialize_host_fields(host, other_host_attributes, staleness_timestamps)
        }
```